src/pipdeptree/_models/dag.py

```diff
@@ -36,22 +36,18 @@ class PackageDAG(Mapping[DistPackage, List[ReqPackage]]):
     """
 
     @classmethod
-    def from_pkgs(cls, pkgs: list[DistInfoDistribution]) -> PackageDAG:
+    def from_pkgs(cls, pkgs: list[Distribution]) -> PackageDAG:
         dist_pkgs = [DistPackage(p) for p in pkgs]
         idx = {p.key: p for p in dist_pkgs}
         m: dict[DistPackage, list[ReqPackage]] = {}
         for p in dist_pkgs:
             reqs = []
-            for r in p.requires():
-                # Requirement key is not sufficiently normalized in pkg_resources - apply additional normalization
-                d = idx.get(pep503_normalize(r.key))
-                # pip's _vendor.packaging.requirements.Requirement uses the exact casing of a dependency's name found in
-                # a project's build config, which is not ideal when rendering.
-                # See https://github.com/tox-dev/pipdeptree/issues/242
-                r.project_name = d.project_name if d is not None else r.project_name
-                pkg = ReqPackage(r, d)
-                reqs.append(pkg)
-            m[p] = reqs
+            if p.requires():
+                for r in p.requires():
+                    d = idx.get(pep503_normalize(r.name))
+                    pkg = ReqPackage(r, d)
+                    reqs.append(pkg)
+                m[p] = reqs
```

Comment:
We perform `m[p] = reqs` even if `p.requires()` is an empty list. So let's just get rid of the if statement here.

Reply:
I I initially used an 'if' statement here because my modification caused DistPackage.requires to return None. It would be better to change the return value to [], which would eliminate the need for the 'if' statement.